### ISSUE TITLE
Document the public complex AD grad/VJP convention

### ISSUE THREAD

**Issue by shinaoka** [tensor4all/tenferro-rs#994]
## Background

`origin/main` currently documents the complex AD convention in `docs/spec/ad-contract.md`, but that file is an internal/deep contract. Users who compare tenferro against JAX will likely look at public `grad`, `vjp`, eager `backward`, tutorials, or API docs first, and the current behavior can look like a conjugation bug unless the cotangent convention is explained up front.

The important point is that tenferro's current VJP convention is internally consistent but not the same public cotangent encoding as JAX's `jax.grad` / `jax.vjp` presentation.

## Current behavior to document

For complex AD, tenferro/tidu currently treats reverse-mode cotangents under the Hermitian real inner product:

```text
<a, b> = Re(conj(a) * b)
```

For a holomorphic scalar map `y = f(z)`:

```text
JVP: dy = f'(z) * dz
VJP: dz_bar = y_bar * conj(f'(z))
```

This means public `grad` with seed `1` returns the Hermitian-adjoint cotangent. Compared with JAX's public `grad` value, the result is conjugated in the common scalar cases:

```text
grad_tenferro(f) = conj(grad_jax(f))
```

Examples that should be documented:

| Function | JAX public `grad` | tenferro current `grad` |
| --- | --- | --- |
| `f(z) = exp(z)` with holomorphic complex output | `exp(z)` | `conj(exp(z))` |
| `f(z) = z^2` with holomorphic complex output | `2z` | `2conj(z)` |
| `L(z) = abs(z)^2`, real-valued loss | `2conj(z)` | `2z` |

This is not a local AD rule failure. It follows from the cotangent encoding. The issue is discoverability and user-facing explanation.

## Proposed documentation updates

Add user-facing documentation outside the deep AD contract, for example:

- A guide page such as `docs/guides/complex-ad.md` or `docs/getting-started/complex-ad.md`.
- A short public-facing note in API docs for `TracedTensorAdExt::grad`, `AdContext::grad`, `vjp`, and eager `EagerTensor::backward` / `grad`.
- A comparison section named something like "Comparison with JAX" that explains:
  - JVPs are unambiguous and match real-linear directional derivatives.
  - VJP/grad values depend on cotangent encoding.
  - tenferro currently uses the Hermitian real-inner-product convention.
  - JAX's public `grad` returns the conjugate representation for these common scalar cases.
  - To compare with JAX scalar `grad`, use `conj(tenferro_grad)`.
- A note in getting-started docs near `PyTorch and JAX Mapping`, since users coming from JAX are likely to look there.
- Minimal runnable examples for:
  - `grad(sum(exp(z)))`
  - `grad(sum(abs(z)^2))`
  - `vjp` with a non-real complex cotangent seed, showing how seed conjugation affects comparison.

## Acceptance criteria

- Users can find the complex AD convention without reading `docs/spec/ad-contract.md`.
- The docs clearly distinguish internal transpose-rule correctness from public `grad` value comparisons.
- The docs include the conversion rule for comparing with JAX:

```text
jax_like_grad = conj(tenferro_grad)
```

for the scalar seed-`1` cases where the two systems are otherwise computing the same real-linear derivative.
- The docs avoid saying simply "JAX-style" without qualifying the cotangent encoding, because that phrase is currently ambiguous.
- At least one public API docstring mentions the convention or links to the new guide.
